drivers/block-nbd.c

```diff
@@ -663,6 +736,81 @@ tdnbd_nbd_negotiate(struct tdnbd_data *prv, td_driver_t *driver)
 	return 0;
 }
 
+static int
+tdnbd_nbd_negotiate_new(struct tdnbd_data *prv, td_driver_t *driver)
+{
+	int rc;
+	uint64_t magic_version;
+	uint16_t gflags;
+	uint32_t cflags = htobe32(NBD_FLAG_FIXED_NEWSTYLE | NBD_FLAG_NO_ZEROES);
+	int sock = prv->socket;
+	#define RECV_BUFFER_SIZE 256
```

Comment:
We seem to have a few of these. Can they be merged?

Reply:
fixed